### PR TITLE
added authentication module

### PR DESCRIPTION
**Pull request by Andrei-Suta**

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Body, Controller, Post } from "@nestjs/common";
+import { User } from "src/user/user.entity";
```
Comment (mgandore) [inline src/auth/auth.controller.ts:3]: new line

File: src/auth/jwt.strategy.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Injectable } from "@nestjs/common";
+import { PassportStrategy } from "@nestjs/passport";
```
Comment (mgandore) [inline src/auth/jwt.strategy.ts:2]: new line for user.entity

File: src/auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { HttpException, HttpStatus, Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { JwtService } from "@nestjs/jwt";
+import * as bcrypt from "bcrypt";
+
+import { User } from "src/user/user.entity";
+import { Repository } from "typeorm";
+import { AuthDTO } from "./auth.dto";
+import { ConfigService } from "@nestjs/config";
```
Comment (mgandore) [inline src/auth/auth.service.ts]: same

**Review by mgandore — APPROVED** [review]

File: src/auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+		user.password = await this.hashPassword(user);
+		return await this.usersRepository.save(user);
+	}
+
+	public async login(authDTO: AuthDTO): Promise<string | never> {
+		const user: User = await this.usersRepository.findOne({ where: { email: authDTO.email } });
+		if (!user) {
+			throw new HttpException("No account registered with the provided email", HttpStatus.NOT_FOUND);
+		}
+		const isMatch = await bcrypt.compare(authDTO.password, user.password);
```
Comment (danielursu) [inline src/auth/auth.service.ts]:
```suggestion
		const isMatch: boolean = await bcrypt.compare(authDTO.password, user.password);
```

File: src/auth/jwt.strategy.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+			jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+			secretOrKey: jwtConstants.secret,
+			ignoreExpiration: false,
+		});
+	}
+	private validate(payload: any): Promise<User | never> {
+		return this.userRepository.findOne({
+			where: { id: payload.id }
+		}
+		);
```
Comment (danielursu) [inline src/auth/jwt.strategy.ts]:
```suggestion
		});
```

**Review by danielursu — APPROVED** [review]

**Pull request merged**